libs/kotlinlib/src/mill/kotlinlib/ksp/Ksp2Module.scala

```diff
@@ -0,0 +1,152 @@
+package mill.kotlinlib.ksp
+
+import coursier.core.VariantSelector.VariantMatcher
+import coursier.params.ResolutionParams
+import mill.*
+import mill.api.Result
+import mill.api.{PathRef, Task}
+import mill.kotlinlib.{Dep, DepSyntax, KotlinModule}
+import mill.util.Jvm
+
+import java.io.File
+
+/**
+ * Sets up the kotlin compiler for using KSP (Kotlin Symbol Processing)
+ * by plugging in the symbol-processing and symbol-processing-api dependencies.
+ *
+ * Use of kotlin-compiler-embedded is also recommended (and thus enabled by default)
+ * to avoid any classpath conflicts between the compiler and user defined plugins!
+ */
+@mill.api.experimental
+trait Ksp2Module extends KotlinModule { outer =>
```

Comment:
Looks like lots of code duplication between `KspModule` and `Ksp2Module`. Probably worth to think about moving those shared tasks in a `KspBaseModule` trait.

Reply:
It makes sense, I need to iterate on it and see how it comes along. It will probably start to get complicated for me when I start testing with Hilt, so I'll probably break the work into multiple PRs and I'd like to keep the 2 separate if I think it's easier to get to the end result, and consolidate towards the end. But I'm not sure yet, just sharing my thoughts